core/blockly.js

```diff
@@ -213,13 +213,18 @@ Blockly.onKeyDown_ = function(e) {
     // hidden.
     return;
   }
+
+
   var deleteBlock = false;
   if (e.keyCode == 27) {
     // Pressing esc closes the context menu.
     Blockly.hideChaff();
-    if (Blockly.keyboardAccessibilityMode) {
-      Blockly.navigation.onKeyPress(e);
+    if (Blockly.keyboardAccessibilityMode && Blockly.navigation.onKeyPress(e)) {
+      return;
```

Comment:
Not sure I fully understand why we need this in the ESC path as well?

Reply:
I think the main reason is that when we hit escape on the toolbox we also need to be able to move the cursor to the workspace.